sdks/go/README.md

```diff
@@ -84,56 +145,237 @@ sentence: 1
+    gcloud storage cat "gs://$GCS_BUCKET/output*" | head
+    ```
+
+    You should see something similar to the following:
+    ```
+    feature: 1
+    block: 1
+    Cried: 1
+    scatter'd: 1
+    she: 44
+    sudden: 1
+    silly: 1
+    More: 6
+    out: 68
+    believe: 3
+    ```
+
+#### Troubleshooting tips
+
+If you get the following error:
+```
+googleapi: Error 400: User project specified in the request is invalid.
+```
+
+Try the following:
+1. Make sure you followed the [Google Cloud Setup](#google-cloud-setup) above.
+2. Configure the [gcloud](https://cloud.google.com/sdk/docs/install-sdk) with your project:
+   ```
+   gcloud config set project <your-project>
+   ```
+3. Consider re-running BOTH `gcloud auth login` and `gcloud auth application-default login`
+   commands.
+
+## Testing
+
+### Requirements (For runner validations)
+
+Below lists additional requirements to execute tests in this repository on your local machine.
+
+#### 1. Java and Python
+
+You **do not** need to know or care about Java or Python to use the Go SDK for your data processing goals.
+
+Java is **only** required to execute any [gradle](https://gradle.org/) commands configured in the
+[Beam](https://github.com/apache/beam) repository. It will be obvious whether you will execute gradle commands in
+sections below. You **do not** need to install [gradle](https://gradle.org/) and simply use the enclosed
+`$BEAM_ROOT/gradlew` executable available at the root of the [Beam](https://github.com/apache/beam) repository.
+
+Python is **only** required if you need to validate tests against the
+[Portable Python Runner](#portable-python-runner). If you are not testing against this
+runner, ignore the Python requirement.
+
+#### 2. Docker
+
+[Docker](https://www.docker.com/) is required for certain but not all runners [See definition](#runner).
+
+As of this writing, [colima](https://github.com/abiosoft/colima), a preferred docker alternative for some developers,
+did not work.
+
+#### 3. Flock
+
+[sdks/go/run_with_go_version.sh](run_with_go_version.sh) requires the use of
+[flock](https://github.com/discoteq/flock).
+
+### Execution
+
+#### 1. Navigate to the go.mod directory
+
+Open a terminal and navigate into the go.mod containing directory of the Beam repository. (See above for what
+`$BEAM_ROOT` means). Notice that you are entering the `$BEAM_ROOT/sdks` and not `$BEAM_ROOT/sdks/go`.
+
+```sh
+cd $BEAM_ROOT/sdks
+```
+
+#### 2. Run Go test
+
+Run go test as you would any Go project.
 
+For unit tests in the exported `pkg/beam` package:
 ```
-$  go run wordcount.go --runner=dataflow --project=<YOUR_GCP_PROJECT> --region=<YOUR_GCP_REGION> --staging_location=<YOUR_GCS_LOCATION>/staging --worker_harness_container_image=<YOUR_SDK_HARNESS_IMAGE_LOCATION> --output=<YOUR_GCS_LOCATION>/output
+go test ./go/pkg/beam...
 ```
 
-The output is a GCS file in this case:
+For integration, load, and regression tests:
+```
+go test ./go/test/...
+```
+
+### Runner validations
```

Comment:
This section above should be either refering to the cwiki page on adding integration tests to the SDK, or replicating that content. Contributors should be aware that gradle commands will run those test suites on the various runners.  

TBH what we have here feels like something that should be mentioned in a language agnostic matter in the Beam Contribution Guide, and this section should refer to gradle commands that execute the validation on each runner. We should not try to justtify the beam infra architectural choices here. We need to assert that they exist, and how contributors use them.